qtox audio/video configuration menu causes a crash

As I wrote in the subject, when you try to configure the audio and video from the specific menu, qTox dies after a few seconds spent in the menu ("Audio/Video" menu).

run on Fedora 21 with XFCE

[update]
i hope that can be useful:

``` bash
warning: GStreamer: unable to query width of frame; no frame grabbed yet (/builddir/build/BUILD/opencv-2.4.9/modules/highgui/src/cap_gstreamer.cpp:809)
warning: GStreamer: unable to query height of frame; no frame grabbed yet (/builddir/build/BUILD/opencv-2.4.9/modules/highgui/src/cap_gstreamer.cpp:823)
warning: GStreamer: unable to query width of frame; no frame grabbed yet (/builddir/build/BUILD/opencv-2.4.9/modules/highgui/src/cap_gstreamer.cpp:809)
warning: GStreamer: unable to query height of frame; no frame grabbed yet (/builddir/build/BUILD/opencv-2.4.9/modules/highgui/src/cap_gstreamer.cpp:823)
warning: GStreamer: unable to query width of frame; no frame grabbed yet (/builddir/build/BUILD/opencv-2.4.9/modules/highgui/src/cap_gstreamer.cpp:809)
warning: GStreamer: unable to query height of frame; no frame grabbed yet (/builddir/build/BUILD/opencv-2.4.9/modules/highgui/src/cap_gstreamer.cpp:823)
warning: GStreamer: unable to query width of frame; no frame grabbed yet (/builddir/build/BUILD/opencv-2.4.9/modules/highgui/src/cap_gstreamer.cpp:809)
warning: GStreamer: unable to query height of frame; no frame grabbed yet (/builddir/build/BUILD/opencv-2.4.9/modules/highgui/src/cap_gstreamer.cpp:823)
warning: GStreamer: unable to query width of frame; no frame grabbed yet (/builddir/build/BUILD/opencv-2.4.9/modules/highgui/src/cap_gstreamer.cpp:809)
warning: GStreamer: unable to query height of frame; no frame grabbed yet (/builddir/build/BUILD/opencv-2.4.9/modules/highgui/src/cap_gstreamer.cpp:823)
warning: GStreamer: unable to query width of frame; no frame grabbed yet (/builddir/build/BUILD/opencv-2.4.9/modules/highgui/src/cap_gstreamer.cpp:809)
warning: GStreamer: unable to query height of frame; no frame grabbed yet (/builddir/build/BUILD/opencv-2.4.9/modules/highgui/src/cap_gstreamer.cpp:823)
warning: GStreamer: unable to query width of frame; no frame grabbed yet (/builddir/build/BUILD/opencv-2.4.9/modules/highgui/src/cap_gstreamer.cpp:809)
warning: GStreamer: unable to query height of frame; no frame grabbed yet (/builddir/build/BUILD/opencv-2.4.9/modules/highgui/src/cap_gstreamer.cpp:823)
warning: GStreamer: unable to query width of frame; no frame grabbed yet (/builddir/build/BUILD/opencv-2.4.9/modules/highgui/src/cap_gstreamer.cpp:809)
warning: GStreamer: unable to query height of frame; no frame grabbed yet (/builddir/build/BUILD/opencv-2.4.9/modules/highgui/src/cap_gstreamer.cpp:823)
warning: GStreamer: unable to query width of frame; no frame grabbed yet (/builddir/build/BUILD/opencv-2.4.9/modules/highgui/src/cap_gstreamer.cpp:809)
warning: GStreamer: unable to query height of frame; no frame grabbed yet (/builddir/build/BUILD/opencv-2.4.9/modules/highgui/src/cap_gstreamer.cpp:823)
warning: GStreamer: unable to query width of frame; no frame grabbed yet (/builddir/build/BUILD/opencv-2.4.9/modules/highgui/src/cap_gstreamer.cpp:809)
warning: GStreamer: unable to query height of frame; no frame grabbed yet (/builddir/build/BUILD/opencv-2.4.9/modules/highgui/src/cap_gstreamer.cpp:823)
warning: GStreamer: unable to query width of frame; no frame grabbed yet (/builddir/build/BUILD/opencv-2.4.9/modules/highgui/src/cap_gstreamer.cpp:809)
warning: GStreamer: unable to query height of frame; no frame grabbed yet (/builddir/build/BUILD/opencv-2.4.9/modules/highgui/src/cap_gstreamer.cpp:823)
warning: GStreamer: unable to query width of frame; no frame grabbed yet (/builddir/build/BUILD/opencv-2.4.9/modules/highgui/src/cap_gstreamer.cpp:809)
warning: GStreamer: unable to query height of frame; no frame grabbed yet (/builddir/build/BUILD/opencv-2.4.9/modules/highgui/src/cap_gstreamer.cpp:823)
warning: GStreamer: unable to query width of frame; no frame grabbed yet (/builddir/build/BUILD/opencv-2.4.9/modules/highgui/src/cap_gstreamer.cpp:809)
warning: GStreamer: unable to query height of frame; no frame grabbed yet (/builddir/build/BUILD/opencv-2.4.9/modules/highgui/src/cap_gstreamer.cpp:823)
warning: GStreamer: unable to query width of frame; no frame grabbed yet (/builddir/build/BUILD/opencv-2.4.9/modules/highgui/src/cap_gstreamer.cpp:809)
warning: GStreamer: unable to query height of frame; no frame grabbed yet (/builddir/build/BUILD/opencv-2.4.9/modules/highgui/src/cap_gstreamer.cpp:823)
warning: GStreamer: unable to query width of frame; no frame grabbed yet (/builddir/build/BUILD/opencv-2.4.9/modules/highgui/src/cap_gstreamer.cpp:809)
warning: GStreamer: unable to query height of frame; no frame grabbed yet (/builddir/build/BUILD/opencv-2.4.9/modules/highgui/src/cap_gstreamer.cpp:823)
warning: GStreamer: unable to query width of frame; no frame grabbed yet (/builddir/build/BUILD/opencv-2.4.9/modules/highgui/src/cap_gstreamer.cpp:809)
warning: GStreamer: unable to query height of frame; no frame grabbed yet (/builddir/build/BUILD/opencv-2.4.9/modules/highgui/src/cap_gstreamer.cpp:823)
warning: GStreamer: unable to query width of frame; no frame grabbed yet (/builddir/build/BUILD/opencv-2.4.9/modules/highgui/src/cap_gstreamer.cpp:809)
warning: GStreamer: unable to query height of frame; no frame grabbed yet (/builddir/build/BUILD/opencv-2.4.9/modules/highgui/src/cap_gstreamer.cpp:823)
CameraWorker: Resolutions () 
OpenCV Error: Unspecified error (GStreamer: unable to start pipeline
) in icvStartPipeline, file /builddir/build/BUILD/opencv-2.4.9/modules/highgui/src/cap_gstreamer.cpp, line 385
terminate called after throwing an instance of 'cv::Exception'
  what():  /builddir/build/BUILD/opencv-2.4.9/modules/highgui/src/cap_gstreamer.cpp:385: error: (-2) GStreamer: unable to start pipeline
 in function icvStartPipeline
```

this is what i see on shell when i try to configure audio and video!

Mine crashes too, fedora21 and qtox-1.0-1462.i386. No camera attached into system.
Last rows from stace:
 what():  /builddir/build/BUILD/opencv-2.4.9/modules/highgui/src/cap_gstreamer.cpp:385: error: (-2) GStreamer: unable to start pipeline
 in function icvStartPipeline

--- a/src/video/cameraworker.cpp
+++ b/src/video/cameraworker.cpp
@@ -146,7 +146,21 @@ void CameraWorker::subscribe()
         if (!cam.isOpened())
         {
             queue.clear();
-            cam.open(camIndex);
+            bool bSuccess = false;
+
+            try
+            {
+                bSuccess = cam.open(camIndex);
+            }
+            catch( cv::Exception& e )
+            {
+                qDebug() << "CameraWorker:" << "OpenCV exception caught: " << e.what();
+            }
+
+            if(!bSuccess)
+            {
+                qDebug() << "CameraWorker: Could not open camera";
+            }
             applyProps(); // restore props
         }
     }
@@ -166,7 +180,20 @@ void CameraWorker::doWork()
     if (!cam.isOpened())
         return;
 
-    if (!cam.read(frame))
+    bool bSuccess = false;
+
+    try
+    {
+            bSuccess = cam.read(frame);
+    }
+    catch( cv::Exception& e )
+    {
+        qDebug() << "CameraWorker:" << "OpenCV exception caught: " << e.what();;
+        this->clock->stop(); // prevent log spamming
+        qDebug() << "CameraWorker: stopped clock";
+    }
+
+    if (!bSuccess)
     {
         qDebug() << "CameraWorker: Cannot read frame";
         return;

